libraries/AP_Motors/AP_Motors_Class.h

```diff
@@ -45,6 +45,43 @@ class AP_Motors {
         MOTOR_FRAME_DODECAHEXA = 12,
         MOTOR_FRAME_HELI_QUAD = 13,
     };
+
+    // return string corresponding to frame_class
+    virtual const char* get_frame_string()
+    {
+        switch (_active_frame_class) {
```

Comment:
This is that if _active_frame_class corresponds to the allocated class then we *don't need* this switch statement.  There's just a method which returns the string.

This has the advantage of keeping things more self-contained so if you add a frame everything's in the one place.

Reply:
I don't see how to avoid a switch statement in MotorsMatrix, since it supports multiple frame classes and types